spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -321,41 +321,50 @@ private ClientSession initClientSession() throws IOException {
 	}
 
 	private void initClient() throws IOException {
-		if (this.initialized.compareAndSet(false, true)) {
-			if (this.port <= 0) {
-				this.port = SshConstants.DEFAULT_PORT;
+		if (!this.initialized) {
+			synchronized (this) {
+				if (!this.initialized) {
+					doInitClient();
+					this.initialized = true;
```

Comment:
@garyrussell , WDYT about this double-check solution instead?

Reply:
It's marginally better, I suppose, but performance is rarely an issue here because most people cache sessions.

That said, the patch looks more complicated than it really is, review with `?w=1`.